tensorflow/core/kernels/data/dataset_test_base.cc

```diff
@@ -560,5 +560,136 @@ Status DatasetOpsTestBase::AddDatasetInput(
   return Status::OK();
 }
 
+Status DatasetOpsTestBase::CheckDatasetNodeName(
+    const DatasetBase& dataset, const string& expected_dataset_node_name) {
+  EXPECT_EQ(dataset.node_name(), expected_dataset_node_name);
+  return Status::OK();
+}
+
+Status DatasetOpsTestBase::CheckDatasetTypeString(
+    const DatasetBase& dataset, const string& expected_dataset_type_string) {
+  EXPECT_EQ(dataset.type_string(), expected_dataset_type_string);
+  return Status::OK();
+}
+
+Status DatasetOpsTestBase::CheckDatasetOutputDtypes(
+    const DatasetBase& dataset, const DataTypeVector& expected_output_dtypes) {
+  TF_EXPECT_OK(
+      VerifyTypesMatch(dataset.output_dtypes(), expected_output_dtypes));
+  return Status::OK();
+}
+
+Status DatasetOpsTestBase::CheckDatasetOutputShapes(
+    const DatasetBase& dataset,
+    const std::vector<PartialTensorShape>& expected_output_shapes) {
+  TF_EXPECT_OK(
+      VerifyShapesCompatible(dataset.output_shapes(), expected_output_shapes));
+  return Status::OK();
+}
+
+Status DatasetOpsTestBase::CheckDatasetCardinality(const DatasetBase& dataset,
+                                                   int64 expected_cardinality) {
+  EXPECT_EQ(dataset.Cardinality(), expected_cardinality);
+  return Status::OK();
+}
+
+Status DatasetOpsTestBase::CheckDatasetSave(const DatasetBase& dataset) {
```

Comment:
In the future, it would be good to extend this test to also check deserialization of the dataset and comparison of the elements produced by the two datasets. I would make that change in a separate PR though.